src/Tribe/Linked_Posts.php

```diff
@@ -0,0 +1,812 @@
+<?php
+/**
+ * Object used to handle the linking/unlinking of post types for events
+ */
+class Tribe__Events__Linked_Posts {
+	/**
+	 * @var string Meta key prefix for linked posts
+	 */
+	const META_KEY_PREFIX = '_tribe_linked_post_';
+
+	/**
+	 * @var Tribe__Events__Linked_Posts Singleton instance of the class
+	 */
+	public static $instance;
+
+	/**
+	 * @var Tribe__Events__Main Singleton
+	 */
+	public $main;
+
+	/**
+	 * @var array Collection of post types that can be linked with events
+	 */
+	public $linked_post_types;
+
+	/**
+	 * Returns a singleton of this class
+	 *
+	 * @return Tribe__Events__Linked_Posts
+	 */
+	public static function instance() {
+		if ( ! self::$instance ) {
+			self::$instance = new self;
+		}
+
+		return self::$instance;
+	}
+
+	/**
+	 * Constructor!
+	 */
+	public function __construct() {
```

Comment:
Should this be changed to protected/private or else be marked `@internal` (since we'd generally prefer people get access the singleton instance)?